packages/extra/test/reference/DayOfMonthTest.js

```diff
@@ -0,0 +1,602 @@
+/*
+ * @copyright (c) 2022, Philipp Thürwächter & Pattrick Hüper & Michał Sobkiewicz
+ * @copyright (c) 2007-present, Stephen Colebourne & Michael Nascimento Santos
+ * @license BSD-3-Clause (see LICENSE in the root directory of this source tree)
+ */
+
+import {
+    ChronoField,
+    ChronoUnit,
+    Clock,
+    DateTimeException,
+    IsoChronology,
+    IsoFields,
+    LocalDate,
+    LocalTime,
+    Month,
+    MonthDay,
+    NullPointerException,
+    TemporalField,
+    TemporalQueries,
+    UnsupportedTemporalTypeException,
+    ValueRange,
+    YearMonth,
+    ZoneId,
+} from '@js-joda/core';
+
+import '../_init';
+
+import { assertEquals, assertFalse, assertSame, assertThrows } from '../testUtils';
+
+import { DayOfMonth } from '../../src/DayOfMonth';
+
+describe('org.threeten.extra.TestDayOfMonth', () => {
+    const MAX_LENGTH = 31;
+    const TEST = DayOfMonth.of(12);
+
+    class TestingField extends TemporalField {
+        getBaseUnit() {
+            return ChronoUnit.DAYS;
+        }
+
+        getRangeUnit() {
+            return ChronoUnit.MONTHS;
+        }
+
+        range() {
+            return ValueRange.of(1, 28, 31);
+        }
+
+        isDateBased() {
+            return true;
+        }
+
+        isTimeBased() {
+            return false;
+        }
+
+        isSupportedBy(temporal) {
+            return temporal.isSupported(ChronoField.DAY_OF_MONTH);
+        }
+
+        // eslint-disable-next-line no-unused-vars
+        rangeRefinedBy(temporal) {
+            return this.range();
+        }
+
+        getFrom(temporal) {
+            return temporal.getLong(ChronoField.DAY_OF_MONTH);
+        }
+
+        adjustInto(temporal, newValue) {
+            return temporal.with(ChronoField.DAY_OF_MONTH, newValue);
+        }
+    }
+
+    TestingField.INSTANCE = new TestingField();
+
+    //-----------------------------------------------------------------------
+    // now()
+    //-----------------------------------------------------------------------
+    describe('now()', () => {
+        it('test_now', () => {
+            let test = DayOfMonth.now();
+            if (LocalDate.now().dayOfMonth() !== test.value()) {
+                test = DayOfMonth.now();
+            }
```

Comment:
What's the purpose if the if clause? Doing the same assignment again, looks strange.

Reply:
See also:
- https://github.com/js-joda/js-joda/blob/master/packages/core/test/reference/LocalDateTest.js#L118-L132,
- https://github.com/js-joda/js-joda/blob/master/packages/core/test/reference/YearMonthTest.js#L37-L53.

Without mocking (freezing) `Clock.systemDefaultZone()` which `now()` uses, similar tests may be impossible to rewrite to be fully deterministic.

Reply:
Fixed in 878336ab4c1a6fe2cf892bc20a1725bae8628904 (still non-fully-deterministic, but at least correct).